src/library/scala/sys/process/BasicIO.scala

```diff
@@ -213,19 +213,24 @@ object BasicIO {
   }
 
   /** Copy contents of stdin to the `OutputStream`. */
+  @deprecated("no longer used")
   def connectToIn(o: OutputStream): Unit = transferFully(Uncloseable protect stdin, o)
 
   /** Returns a function `OutputStream => Unit` that either reads the content
     * from stdin or does nothing. This function can be used by
     * [[scala.sys.process.ProcessIO]].
     */
+  @deprecated("no longer used")
   def input(connect: Boolean): OutputStream => Unit = { outputToProcess =>
     if (connect) connectToIn(outputToProcess)
     outputToProcess.close()
   }
 
+  /** Just closes given OutputStream, to be passed to ProcessIO. */
+  private[this] def closer(stream: OutputStream): Unit = stream.close()
```

Comment:
IMO, `closer` is not appropriate naming since `closer` just closes `OutputStream`.   For your purpose , it seems that the following definition is better. 

```
private[this] val closer: (OutputStream) => Unit = (stream) => stream.close
```